pkg/cmd/roachtest/metrics/prometheus_test.go

```diff
@@ -0,0 +1,288 @@
+// Copyright 2023 The Cockroach Authors.
+//
+// Use of this software is governed by the CockroachDB Software License
+// included in the /LICENSE file.
+
+package metrics
+
+import (
+	"fmt"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+// basicMetricsAPI is a simplified implementation for testing
+type basicMetricsAPI struct {
+	debugMode      bool
+	lastQuery      string
+	lastQueryValue float64
+}
+
+// basicMetricQuery is a simplified implementation for testing
+type basicMetricQuery struct {
+	api         *basicMetricsAPI
+	metricName  string
+	labels      map[string]string
+	timeRange   string
+	isRateQuery bool
+	aggregation string
+}
+
+func (b *basicMetricsAPI) EnableDebugMode(enabled bool) {
+	b.debugMode = enabled
+}
+
+func (b *basicMetricsAPI) GetLastQuery() string {
+	return b.lastQuery
+}
+
+func (b *basicMetricsAPI) GetLastQueryResult() float64 {
+	return b.lastQueryValue
+}
+
+func (b *basicMetricsAPI) AssertThat(metricName string) MetricQuery {
+	return &basicMetricQuery{
+		api:        b,
+		metricName: metricName,
+		labels:     make(map[string]string),
+	}
+}
+
+func (b *basicMetricsAPI) Close() error {
+	return nil
+}
+
+func (b *basicMetricsAPI) CompareValuesOverTime(
+	metricName string,
+	startTime, endTime time.Time,
+) MetricComparison {
+	// For testing, just store the query
+	b.lastQuery = metricName
+	b.lastQueryValue = 1.0
+	return &basicMetricComparison{api: b}
+}
+
+func (q *basicMetricQuery) ForService(serviceName string) MetricQuery {
+	q.labels["service"] = serviceName
+	return q
+}
+
+func (q *basicMetricQuery) ForNode(nodeID string) MetricQuery {
```

Comment:
For ease of use this should probably take in a `option.NodeOptionList` and do the conversion to string for the user.